src/Microsoft.AspNetCore.Server.Testing/Deployers/NginxDeployer.cs

```diff
@@ -0,0 +1,165 @@
+                {
+                    // process didn't exit
+                    throw new Exception("Failed to test for sudo");
+                }
+
+                if (errorRecieved == true)
+                {
+                    throw new InvalidOperationException("User doesn't have sudo permissions, can't deploy using Nginx");
+                }
+            }
+
+            // stop Nginx so that we can restart it with a new config file
+            startInfo.Arguments = "nginx -s stop";
+            using (var stopNginx = new Process() { StartInfo = startInfo })
+            {
+                stopNginx.ErrorDataReceived += (sender, dataArgs) =>
+                {
+                    if (!string.IsNullOrEmpty(dataArgs.Data))
+                    {
+                        Logger.LogWarning("nginx: " + dataArgs.Data);
+                    }
+                };
+                stopNginx.OutputDataReceived += (sender, dataArgs) =>
+                {
+                    if (!string.IsNullOrEmpty(dataArgs.Data))
+                    {
+                        Logger.LogInformation("nginx: " + dataArgs.Data);
+                    }
+                };
+                stopNginx.Start();
+                stopNginx.BeginErrorReadLine();
+                stopNginx.BeginOutputReadLine();
+                if (stopNginx.WaitForExit(2000) == false)
+                {
+                    // process didn't exit
+                    throw new Exception("Failed to stop Nginx");
+                }
+            }
+
+            // copy nginx.conf template and replace pertinent information
+            var configTemplateContent = File.ReadAllText(Path.Combine(DeploymentParameters.ApplicationPath, "nginx.conf"));
+            configTemplateContent = configTemplateContent.Replace("[user]", Environment.GetEnvironmentVariable("LOGNAME"))
+                .Replace("[listenPort]", originalUri.Port.ToString())
+                .Replace("[redirectUri]", redirectUri);
+            File.WriteAllText(configFile, configTemplateContent);
+
+            startInfo.Arguments = $"nginx -c {configFile}";
+            using (var startNginx = new Process() { StartInfo = startInfo })
+            {
+                startNginx.ErrorDataReceived += (sender, dataArgs) =>
+                {
+                    if (!string.IsNullOrEmpty(dataArgs.Data))
+                    {
+                        Logger.LogWarning("nginx: " + dataArgs.Data);
+                    }
+                };
+                startNginx.OutputDataReceived += (sender, dataArgs) =>
+                {
+                    if (!string.IsNullOrEmpty(dataArgs.Data))
+                    {
+                        Logger.LogInformation("nginx: " + dataArgs.Data);
+                    }
+                };
+                startNginx.Start();
+                startNginx.BeginErrorReadLine();
+                startNginx.BeginOutputReadLine();
+                if (startNginx.WaitForExit(2000) == false)
```

Comment:
What about the exit code from the process. I guess it returns 0 if the server started and non zero otherwise